app/views.py

```diff
@@ -73,15 +73,17 @@ def get(self, request, game_id, player_id):
 
         thumbs_seen, eyes_seen = player_info(game, player)
 
+        roles = [player.role for player in game.player_set.order_by('role').filter(is_kicked=False)]
```

Comment:
unsure if you just haven't changed it yet but any reason you prefer `[player.role for player in game.players().order_by('role')]` to `game.players().order_by('role').values_list('role', flat=True)`?

Also you don't need to do `filter(is_kicked=False)` when you do `game.players()` since that method does it for you (there's never a case where you care about kicked players).

Reply:
i personally just think the list comprehension looks nicer, if u care at all i'll change it

Reply:
Also regarding the kicked thing, i think that's actually false.

Here's what I see if I remove that, kick a player, add another one, then start the game:

![image](https://user-images.githubusercontent.com/7587265/35009921-346cc5a4-facf-11e7-8e7f-f52957543785.png)

Reply:
this is probably just me being confused because i've never used django before -- as far as i can tell, there's kicked filtering when you use the `players()` function, but not when you use `player_set`.

Reply:
but anyways i want to use `player_set` because as far as i can tell it gives me nice ORM functionality like `order_by` that i lose if i use `players()`.

if i'm just confused please let me know

Reply:
(here's what i get when i try to use `players()` instead of `player_set`)

![image](https://user-images.githubusercontent.com/7587265/35010095-c09c2d4e-facf-11e7-97da-5f39e13ff47e.png)

Reply:
I definitely prefer `game.players().order_by('role').values_list('role', flat=True)` (more django less python, cleaner to me).

git pull pls I changed how `.players()` works recently. It returns a queryset now

Reply:
oh oof sorry thanks